stdlib/option.gr

```diff
@@ -351,7 +351,7 @@ provide let or = (optionA, optionB) => {
  *
  * @since v0.2.0
  */
-provide let and = (optionA, optionB) => {
+provide let and_ = (optionA, optionB) => {
```

Comment:
Please open an issue to rename these functions in option & result. We should do `or` while we are at it.